framework/src/base/FlagElementsThread.C

```diff
@@ -57,7 +57,10 @@ void
 FlagElementsThread::onElement(const Elem *elem)
 {
   dof_id_type dof_number = elem->dof_number(_system_number, _field_var_number, 0);
-  Marker::MarkerValue marker_value = (Marker::MarkerValue)_serialized_solution[dof_number];
+
+  // This is basically lround() but doesn't require C99 or C++11
+  Marker::MarkerValue marker_value =
+    (Marker::MarkerValue)(std::ceil(_serialized_solution[dof_number]-0.5));
```

Comment:
We already use C99 `round` in GeneratedMesh.C and there's a `MathUtils::round()` interface which also uses it.  The only difference between `round` and `lround` is that the latter returns a `long int`... so if it's the right thing here we should definitely use it.